### PR TITLE
Run NormalizeScore correctly

### LINKED ISSUE #89: Scores from `NormalizedScore` are not recorded correctly.
ssia... 🤦‍♂️

/kind bug
/assign

### PR DESCRIPTION
**Pull request by sanposhiho**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Scores from NormalizedScore are not recorded correctly, because `simulatorPlugin.ScoreExtensions()` returns internal plugins's `ScoreExtensions()`. We should use `simulatorPlugin.NormalizeScore()` in ScoreExtension, so should return simulatorPlugin itself.

#### Which issue(s) this PR fixes:

Fixes #89

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: scheduler/plugin/plugins.go [hunk 1]
```diff
@@ -270,14 +270,14 @@ func newSimulatorPlugin(s store, p framework.Plugin, weight int32) framework.Plu
 
 func (pl *simulatorPlugin) Name() string { return pl.name }
 func (pl *simulatorPlugin) ScoreExtensions() framework.ScoreExtensions {
-	if pl.originalScorePlugin != nil {
-		return pl.originalScorePlugin.ScoreExtensions()
+	if pl.originalScorePlugin != nil && pl.originalScorePlugin.ScoreExtensions() != nil {
```
Comment (riita10069) [inline scheduler/plugin/plugins.go:273]: Doesn't it become `nil pointer exception` when `pl.originalScorePlugin` is nil?

File: scheduler/plugin/plugins.go [hunk 2]
```diff
@@ -270,14 +270,14 @@ func newSimulatorPlugin(s store, p framework.Plugin, weight int32) framework.Plu
...
+		return pl
 	}
 	return nil
 }
 
 func (pl *simulatorPlugin) NormalizeScore(ctx context.Context, state *framework.CycleState, pod *v1.Pod, scores framework.NodeScoreList) *framework.Status {
-	if pl.originalScorePlugin == nil {
+	if pl.originalScorePlugin == nil || pl.originalScorePlugin.ScoreExtensions() == nil {
```
Comment (riita10069) [inline scheduler/plugin/plugins.go:280]: ditto

Comment (sanposhiho) [reply scheduler/plugin/plugins.go:280]: same
https://github.com/kubernetes-sigs/kube-scheduler-simulator/pull/90#discussion_r797735149

File: scheduler/plugin/plugins.go (hunk 1 above)
Comment (riita10069) [reply scheduler/plugin/plugins.go:273]: It's awesome! I didn't know that. Thank you.

**Pull request merged**